### PR TITLE
Remove duplicate route53 options

### PR DESCRIPTION
**Pull request by Oshawk**
Duplicate route53 option (certbot-route53:auth) removed.

Fixes https://github.com/certbot/certbot/issues/6209

Not sure if it is worth updating the changelog for such a small change.

### CODE REVIEW

**Comment by Oshawk** [comment]
I have been trying to find where it is generated for about 2 hours but have had no luck. The only mention of rout53:auth seems to be in the `setup.py` and removing that does not fix anything. I'm afraid I am stumped.

**Comment by trinopoty** [comment]
Check out `certbot-dns-route53/certbot_dns_route53/authenticator.py`

**Comment by Oshawk** [comment]
Think that may have solved it. I was having trouble due to the egg info.

**Comment by ohemorange** [comment]
Hey @Oshawk, how's progress going?

**Comment by Oshawk** [comment]
Hi @ohemorange I have discovered that I need to somehow pass in argparse.SUPRESS or use add_deprecated_argument() instead of the normal add_argument() although I am not sure how I would do this as arguments are added in bulk I believe. My current plan is to add a line to the cli.py file to test if the argument is the shim and if so supress it. I sadly don't see any cleaner solution at the moment.

**Comment by ohemorange** [comment]
That sounds reasonable enough.

**Comment by trinopoty** [comment]
Removing `'certbot-route53:auth = certbot_dns_route53.authenticator:Authenticator'` from `certbot-dns-route53/setup.py` line 54 removes the output in help text but I'm not sure if that prevents certbot from detecting the plugin entirely.

**Comment by trinopoty** [comment]
There doesn't seem to be any other way to remove it from help text without adding a specific condition for the plugin in cli.py

**Comment by ohemorange** [comment]
> Removing `'certbot-route53:auth = certbot_dns_route53.authenticator:Authenticator'` from `certbot-dns-route53/setup.py` line 54 removes the output in help text but I'm not sure if that prevents certbot from detecting the plugin entirely.

Almost certainly, but you can test this by removing that line, then trying to select the auth plugin from the command line

**Comment by Oshawk** [comment]
@ohemorange The latest commit should do it.

File: certbot/cli.py [hunk 1]
```diff
@@ -836,8 +839,9 @@ def add_plugin_args(self, plugins):
 
         """
         for name, plugin_ep in six.iteritems(plugins):
-            parser_or_group = self.add_group(name,
-                                             description=plugin_ep.long_description)
+            if name != 'certbot-route53:auth':  # Remove duplicate route53 options from help.
+                parser_or_group = self.add_group(name,
+                                                 description=plugin_ep.long_description)
             plugin_ep.plugin_cls.inject_parser_options(parser_or_group, name)
```
Comment (ohemorange) [inline certbot/cli.py]: This should also be in the `if`

File: certbot/cli.py [hunk 2]
```diff
@@ -750,6 +750,9 @@ def add(self, topics, *args, **kwargs):
 
         """
 
+        if args[0] == '--certbot-route53:auth-propagation-seconds':  # Remove duplicate route53 options from help.
```
Comment (ohemorange) [inline certbot/cli.py]: Why is adding it in the other place insufficient?

File: certbot/cli.py (hunk 1 above)
Comment (ohemorange) [reply certbot/cli.py]: `parser_or_group` is set to the previous loop's iteration if the first `if` isn't hit.

Comment (Oshawk) [reply certbot/cli.py]: Yes it is, I believe it is set to apache. If groups are not used for anything other than help then does this matter as it is always hidden? I don;t know if it is possible to add it to no group.

File: certbot/cli.py (hunk 2 above)
Comment (ohemorange) [reply certbot/cli.py]: Adding it to the Apache group isn't ideal.

Found a cleaner way to do this: in `determine_help_topics`, can just always set this topic to false:
```
if chosen_topic == "all":
    return dict([(t, True) if t != 'certbot-route53:auth' else (t, False) for t in self.help_topics])
```

Then it's only one place to keep track of things.

Comment (Oshawk) [reply certbot/cli.py]: Yep, that is certainly better. Have made the commit.

File: docs/cli-help.txt [hunk 3]
```diff
@@ -503,15 +503,6 @@ apache:
                         Full path to Apache control script (default:
                         apache2ctl)
 
-certbot-route53:auth:
```
Comment (ohemorange) [inline docs/cli-help.txt]: This file is automatically generated at release time and shouldn't be updated manually.

File: certbot/cli.py [hunk 4]
```diff
@@ -749,7 +749,6 @@ def add(self, topics, *args, **kwargs):
         :param dict **kwargs: various argparse settings for this argument
 
         """
-
```
Comment (ohemorange) [inline certbot/cli.py]: nit: re-add line

File: docs/cli-help.txt [hunk 5]
```diff
@@ -506,7 +506,6 @@ apache:
 certbot-route53:auth:
   Obtain certificates using a DNS TXT record (if you are using AWS Route53
   for DNS).
-
```
Comment (ohemorange) [inline docs/cli-help.txt]: Add this back and we're good to go!

**Pull request merged**